src/renderer/components/configure-recording-set-page/index.tsx

```diff
@@ -1,112 +1,44 @@
 import log from 'electron-log';
-import React, { FC, Fragment, MouseEventHandler, useContext, useEffect, useState } from 'react';
-import Col from 'react-bootstrap/Col';
-import Container from 'react-bootstrap/Container';
-import Row from 'react-bootstrap/Row';
-import { useTranslation } from 'react-i18next';
-import CSSTransition from 'react-transition-group/CSSTransition';
-import useLocalStorage from 'react-use/lib/useLocalStorage';
+import React, { FC, MouseEventHandler, useState } from 'react';
+import { useLocalStorage, usePrevious } from 'react-use';
 
-import { RecordingProjectContext } from '../../contexts';
-import { PROJECT_CONFIG_FILENAME } from '../../env-and-consts';
-import { Consumer, RecordingProject, RecordingSet, ScaleKey, SupportedOctave } from '../../types';
-import {
-  checkFileExistence,
-  deleteFolder,
-  ensureFolderExists,
-  getLSKey,
-  join,
-  naiveSerialize,
-  readFile,
-  writeFile,
-} from '../../utils';
-import BackButton from '../back-button';
-import NextButton from '../next-button';
-
-import AddRecordingSetButton from './add-recording-set-button';
-import CreatedRecordingSetList from './created-recording-set-list';
-import SetMetaConfiguration from './set-meta-configuration';
-import SetRecordingListConfiguration from './set-recording-list-configuration';
-import { BuiltInRecordingList } from './types';
+import { Consumer, RecordingSet } from '../../types';
+import { checkFileExistence, getLSKey, join } from '../../utils';
 
+import { ConfigureRecordingSet } from './configure-recording-set';
 import './show-details.scss';
-
-interface ProjectFile extends RecordingProject {
-  name: string;
-  recordingSets: RecordingSet[];
-}
-
-const DUMMY_PROJECT_FILE = {
-  name: '',
-  rootPath: '',
-  recordingSets: [],
-};
-
-const BUILT_IN_RECORDING_LISTS: BuiltInRecordingList[] = [
-  'デルタ式英語リストver5 (Delta English Ver. 5)',
-  'Z式CVVC-Normal (Z Chinese CVVC - Normal)',
-];
-
+import './index.scss';
+import { PreviewPage } from './preview-page';
+
+/*
+ Represents the page states controlled by configure-recording-set.
+ 'external' represents pages external from this system
```

Comment:
The configure-recording-set page tracks a local state so it can determine which of it's subpages (the base configure-recording-set page or one of the preview metadata pages).  When the base configure-recording-set page loads it needs to know whether it's being loaded after one of the preview pages or from a page that isn't a local state of configure-recording-set (in order to determine whether to animate the transition in or not). The external state represents this second case

Reply:
apologies, I forgot that I split off the states into two separate types of state.  
There's three 'primary' sub-states: home, external, and metadata.  Home refers to the configure-recording-set base page, external refers to a page outside of the configure-recording-set subsystem, and metadata, which refers to the metadata  preview pages (oto/reclist/dvcfg).  
Since it is determined at runtime which preview pages are able to be accessed (since they do not show up if the file does not exist in the reclist's folder), i needed to be able to iterate though whichever pages existed when the side arrow button was clicked.  In order to do this, i separated the states that could potentially get iterated through into their own type of state with a different type of setter.  So basically, as soon as the "show details" button is clicked, the RecordingPageState is set to metadata, and the MetadataState is what's used to determine what is rendered until the state is set back to home;
When rendering the configure-recording-set base page, it either needs to be told to use an animated transition (if it's coming back from one of the metadata pages) or to appear normally (if it's coming from any other page) When either the "next" or "back" button gets clicked, it sets the state to external so that when we go back to the configure-recording-set base page, it knows not to use the animated transition

Reply:
I see. Thank you!